src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-perf/xunitrunner-perf.py

```diff
@@ -140,17 +156,65 @@ def post_process_perf_results(settings, results_location, workitem_dir):
-        raise Exception('Failed to generate json from csv file')
 
-    perfsettings = copy.deepcopy(settings)
+    log.info('Uploading the results.csv file')
+    _write_output_path(os.path.join(workitem_dir, 'results.csv'), settings)
+
+    perfsettingsjson = ''
     with open(os.path.join(perfscriptsdir, 'xunitrunner-perf.json'), 'rb') as perfsettingsjson:
-        # upload the json using perf-specific details
+        # read the perf-specific settings
         perfsettingsjson = json.loads(perfsettingsjson.read())
-        perfsettings.output_uri = perfsettingsjson['RootURI']
-        perfsettings.output_write_token = perfsettingsjson['WriteToken']
-        perfsettings.output_read_token = perfsettingsjson['ReadToken']
-        _write_output_path(jsonPath, perfsettings)
+
+    jsonArgsDict = dict()
+    jsonArgsDict['--csvFile'] = os.path.join(workitem_dir, 'results.csv')
+    jsonArgsDict['--jsonFile'] = jsonPath
+    jsonArgsDict['--jobName'] =  settings.correlation_id
+    jsonArgsDict['--jobDescription'] = '...'
+    jsonArgsDict['--configName'] = perfsettingsjson['TargetQueue']
+    jsonArgsDict['--jobGroupName'] = 'CoreFX'
+    jsonArgsDict['--jobTypeName'] = 'Private'
+    jsonArgsDict['--username'] = perfsettingsjson['Creator']
+    jsonArgsDict['--userAlias'] = perfsettingsjson['Creator']
+    jsonArgsDict['--branch'] = perfsettingsjson['TestProduct']
+    jsonArgsDict['--buildInfoName'] = perfsettingsjson['BuildMoniker']
+
+    # extract build number from buildmoniker if official build
+    buildtokens = perfsettingsjson['BuildMoniker'].split('-')
+    if len(buildtokens) < 3:
+        jsonArgsDict['--buildNumber'] = perfsettingsjson['BuildMoniker']
+    else:
+        jsonArgsDict['--buildNumber'] = buildtokens[len(buildtokens)-2] +'.'+buildtokens[len(buildtokens)-1]
+
+    jsonArgsDict['--machinepoolName'] = perfsettingsjson['TargetQueue']
+    jsonArgsDict['--machinepoolDescription'] = '...'
+    jsonArgsDict['--microarchName'] = 'SSE2' # cannot be obtained by cpu-info; need to figure out some other way
+    jsonArgsDict['--numberOfCores'] = psutil.cpu_count(logical=False)
+    jsonArgsDict['--numberOfLogicalProcessors'] = psutil.cpu_count(logical=True)
+    # psutil returns mem in bytes, convert it to MB for readability
+    jsonArgsDict['--totalPhysicalMemory'] = psutil.virtual_memory().total/1024
+    jsonArgsDict['--osInfoName'] = platform.system()
+    jsonArgsDict['--osVersion'] = platform.version()
+    jsonArgsDict['--machineName'] = platform.node()
+
+    info = cpuinfo.get_cpu_info()
+    jsonArgsDict['--architectureName'] = format(info['arch'])
+    jsonArgsDict['--machineDescription'] = format(info['brand'])
+    jsonArgsDict['--manufacturerName'] = format(info['vendor_id'])
+
+    jsonCmd = '%HELIX_PYTHONPATH% '+os.path.join(perfscriptsdir, 'csvjsonconvertor.py')
+    for key, value in jsonArgsDict.iteritems():
+        jsonCmd = jsonCmd+' '+key+' "'+str(value)+'"'
+
+    if os.system(jsonCmd) != 0:
```

Comment:
You could just load the other python module instead of calling back out to the system.